llvmlite/ir/module.py

```diff
@@ -31,6 +31,17 @@ def add_metadata(self, operands):
             md = self._metadatacache[key]
         return md
 
+    def add_debug_info(self, kind, operands, is_distinct=False):
+        n = len(self.metadata)
+        operands = tuple((key, operands[key]) for key in operands)
```

Comment:
There is *a lot* of debug metadata emitted, especially if your frontend, like mine, discerns column numbers. One DILocation per a few instructions.

In any case llvmlite is already nondeterministic, since it uses dicts and hashing in many places (like for metadata itself), and there's something going on with types too that makes the `.number` suffixes change between runs. Another PR may address this, maybe even mine if that type naming issue annoys me enough.